csharp/SecureMemory/SecureMemory/ProtectedMemoryImpl/Libc/LibcProtectedMemoryAllocatorLP64.cs

```diff
@@ -18,15 +18,14 @@ internal abstract class LibcProtectedMemoryAllocatorLP64 : IProtectedMemoryAlloc
         protected LibcProtectedMemoryAllocatorLP64(LibcLP64 libc)
         {
             this.libc = libc ?? throw new ArgumentNullException(nameof(libc));
-
-            libc.getrlimit(GetMemLockLimit(), out var rlim);
-            if (rlim.rlim_max == rlimit.UNLIMITED)
+            var rlim = GetMemlockResourceLimit();
+            if (rlim == rlimit.UNLIMITED || rlim > long.MaxValue)
```

Comment:
This is a signed / unsigned comparison for the second value.  I think it's okay and you have a test that covers that path, but I'm making a note of it.